src/runtime/object.cpp

```diff
@@ -1467,6 +1472,17 @@ extern "C" LEAN_EXPORT double lean_float_scaleb(double a, b_lean_obj_arg b) {
    }
 }
 
+extern "C" LEAN_EXPORT uint8_t lean_float_isnan(double a) { return (std::isnan)(a); }
+extern "C" LEAN_EXPORT uint8_t lean_float_isfinite(double a) { return (std::isfinite)(a); }
+extern "C" LEAN_EXPORT uint8_t lean_float_isinf(double a) { return (std::isinf)(a); }
+extern "C" LEAN_EXPORT obj_res lean_float_frexp(double a) {
+    object* r = lean_alloc_ctor(0, 2, 0);
+    int exp;
+    lean_ctor_set(r, 0, lean_box_float((std::frexp)(a, &exp)));
```

Comment:
https://en.cppreference.com/w/cpp/numeric/math/frexp#Error_handling

> - If arg is ±∞, it is returned, and an unspecified value is stored in *exp.
> - If arg is NaN, NaN is returned, and an unspecified value is stored in *exp.

Probably it should check if `a` is finite.